### DISCUSSION TITLE
Upload and Transcription Progress Monitoring

### DISCUSSION THREAD

**Discussion by rasif-ali** [deepgram/community#819] [General help] [answered]
Hello Deepgram community,

I could not find documentation on how to monitor upload and transcription progress when handling large pre-recorded files via deepgram API. 

1. Upload Progress: How can I monitor the progress of large file uploads to your API?
2. Transcription Progress: Is there a way to check the progress of the transcription process once a file has been uploaded?

Having visibility into these processes would greatly enhance integration with Deepgram for handling large audio files efficiently.

Thank you for your assistance!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment, marked as answer]
Hi @rasif-ali ,

The short answer is, no, we don't provide progress monitoring in the way you are requesting. It's an interesting idea and I will pass it on to the product team as a suggestion for a feature to offer.

Here's a longer answer (and probably not entirely satisfying)...

Some libraries or tools provide progress events that you can listen to. As for monitoring the transcription progress, Deepgram does not provide real-time progress updates for transcription of pre-recorded files. However, there are a couple of ways to know when the transcription is complete:
- AWS S3 Event Notifications: If you're using AWS S3 for storing your transcripts, you can set up S3 Event Notifications. These notifications can trigger an event (like a Lambda function) when a file (in this case, the transcript) is uploaded to a certain bucket/location. This way, your application can know when the transcription is complete.
- Deepgram's Callback Feature: When requesting a transcript from Deepgram, you can use the callback feature. This allows you to redirect any transcription results to the URL of your choice. This way, you can build any additional business logic to run once you have the transcription data.

Thank you for your question/suggestion; I'll definitely share it with the team.

Sources:
https://developers.deepgram.com/docs/using-aws-s3-presigned-urls-with-the-deepgram-api#responding-to-transcript-completion-using-s3-event-notifications
https://developers.deepgram.com/docs/using-callbacks-to-return-transcripts-to-your-server
